### PR TITLE
Remove `interface.run_conversion` without nwbfile_path

### PR DESCRIPTION
**Pull request by h-mayorquin**
The deadline for the deprecation has expired.

### CODE REVIEW

File: src/neuroconv/basedatainterface.py [hunk 1]
```diff
@@ -116,7 +115,7 @@ def add_to_nwbfile(self, nwbfile: NWBFile, **conversion_options) -> None:
 
     def run_conversion(
         self,
-        nwbfile_path: Optional[str] = None,
+        nwbfile_path: Union[str, Path],
```
Comment (CodyCBakerPhD) [inline src/neuroconv/basedatainterface.py:118]: Do we actually support `Path` here generally yet? IDK if PyNWB/HDMF/HDMF-Zarr want `str` only at IO time

Comment (h-mayorquin) [reply src/neuroconv/basedatainterface.py:118]: I have always assumed we do. PYNWB does not but I am less sure about zarr. Let's revert to restrictive then just out of caution.

Comment (h-mayorquin) [reply src/neuroconv/basedatainterface.py]: Done.

**Comment by CodyCBakerPhD** [comment]
Looks like a few straggler tests that still used the deprecated approach😆

**Review by CodyCBakerPhD — APPROVED** [review]

**Pull request merged**